### PR TITLE
stPreCompiledContracts :yul fix

### PR DESCRIPTION
**Pull request by qbzzt**

### CODE REVIEW

File: src/GeneralStateTestsFiller/stPreCompiledContracts/precompsEIP2929Filler.yml [hunk 1]
```diff
@@ -317,19 +335,11 @@ precompsEIP2929:
...
-    - :label no  :abi f(uint,uint) 0x10c0de 0xf100
-
-    # Account that has already been touched
-    - :label yes :abi f(uint,uint) 0x101157 0xf100
 
     # When we do send wei there is an extra cost for the first access because it
     # needs to create the trie for the account. This is not an issue in the real
     # world because on mainnet these tries already exist (each of these accounts
-    # has at least one wei, but to 0x0...020
+    # has at least one wei, up to 0x0...020
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stPreCompiledContracts/precompsEIP2929Filler.yml]: can make up to 0x0...100 for future precompiles

File: src/GeneralStateTestsFiller/stPreCompiledContracts/precompsEIP2929Filler.yml [hunk 2]
```diff
@@ -298,6 +304,18 @@ precompsEIP2929:
     # The first parameter is the address to call
     # The second parameter is the action
 
+
+    # Empty account (nothing there)
+    - :label no  :abi f(uint,uint) 0x100000 0xf100
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stPreCompiledContracts/precompsEIP2929Filler.yml:315]: what does no means why it changed to yes? need to be more descriptive

File: src/GeneralStateTestsFiller/stPreCompiledContracts/precompsEIP2929Filler.yml [hunk 3]
```diff
@@ -255,6 +256,9 @@ precompsEIP2929:
           
           sstore(0, eq(gasTest1, gasTest2))
           sstore(1, sub(gasTest1, gasTest2))
+
+          // Otherwise a Yul optimization throws us off
+          let noOptimization := msize()
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stPreCompiledContracts/precompsEIP2929Filler.yml:261]: calling msize() disables optimization? strange

File: src/GeneralStateTestsFiller/stPreCompiledContracts/idPrecompsFiller.yml [hunk 4]
```diff
@@ -397,9 +405,12 @@ idPrecomps:
       gas: !!int -1
       value: !!int -1
     network: 
-    - ">=Istanbul"
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stPreCompiledContracts/idPrecompsFiller.yml:400]: must remain >=

File: src/GeneralStateTestsFiller/stPreCompiledContracts/precompsEIP2929Filler.yml (hunk 1 above)
Comment (qbzzt) [reply src/GeneralStateTestsFiller/stPreCompiledContracts/precompsEIP2929Filler.yml]: I decided to change that part of the comment. When I wrote it I thought mainnet is the only one that mattered, but now we have rollups where precompiles may not have a trie.

File: src/GeneralStateTestsFiller/stPreCompiledContracts/precompsEIP2929Filler.yml (hunk 2 above)
Comment (qbzzt) [reply src/GeneralStateTestsFiller/stPreCompiledContracts/precompsEIP2929Filler.yml:315]: Clarified

File: src/GeneralStateTestsFiller/stPreCompiledContracts/idPrecompsFiller.yml (hunk 4 above)
Comment (qbzzt) [reply src/GeneralStateTestsFiller/stPreCompiledContracts/idPrecompsFiller.yml:400]: I did that for the `:label yes` and `:label no`.

I added a third label, `:label from_cancun`, for the precompile that cancun adds.

**Pull request merged**